rust/src/measurements/make_private_expr/expr_laplace/mod.rs

```diff
@@ -0,0 +1,320 @@
+    {
+        let scale = RBig::try_from(scale)
+            .map_err(|_| PolarsError::ComputeError("scale must be finite".into()))?;
+        Ok(series
+            // unpack the series into a chunked array
+            .unpack::<PT>()?
+            // apply Laplace noise to the non-null values
+            .try_apply_nonnull_values_generic(|v| {
+                let v = IBig::from(v);
+                sample_discrete_laplace(scale.clone())
+                    .map(|noise| PT::Physical::saturating_cast(v + noise))
+            })?
+            // convert the resulting chunked array back to a series
+            .into_series())
+    }
+
+    // PT stands for Polars Type
+    fn laplace_impl_float<PT: 'static + PolarsDataType>(
+        series: &Series,
+        scale: f64,
+    ) -> PolarsResult<Series>
+    where
+        // the physical (rust) dtype must be a float
+        for<'a> PT::Physical<'a>: Float + InfCast<f64>,
+        // for calibration of the discretization constant k
+        for<'a> i32: ExactIntCast<<PT::Physical<'a> as FloatBits>::Bits>,
+        // must be able to construct the chunked array corresponding to the physical dtype from an iterator
+        for<'a> PT::Array:
+            ArrayFromIter<PT::Physical<'a>> + ArrayFromIter<Option<PT::Physical<'a>>>,
+        // must be able to convert the chunked array to a series
+        ChunkedArray<PT>: IntoSeries,
+    {
+        // cast to the physical (rust) dtype (either f32 or f64)
+        let scale = PT::Physical::inf_cast(scale)?;
+        let k = get_discretization_consts::<PT::Physical<'static>>(None)?.0;
+        Ok(series
+            // unpack the series into a chunked array
+            .unpack::<PT>()?
+            // apply Laplace noise to the non-null values
+            .try_apply_nonnull_values_generic(|v| sample_discrete_laplace_Z2k(v, scale, k))?
+            // convert the resulting chunked array back to a series
+            .into_series())
+    }
+
+    match series.dtype() {
+        // the u8 and u16 features are not enabled in the Polars crate dependency features
+        // DataType::UInt8 => laplace_impl_integer::<UInt8Type>(series, scale),
+        // DataType::UInt16 => laplace_impl_integer::<UInt16Type>(series, scale),
+        DataType::UInt32 => laplace_impl_integer::<UInt32Type>(series, scale),
+        DataType::UInt64 => laplace_impl_integer::<UInt64Type>(series, scale),
+        DataType::Int8 => laplace_impl_integer::<Int8Type>(series, scale),
+        DataType::Int16 => laplace_impl_integer::<Int16Type>(series, scale),
+        DataType::Int32 => laplace_impl_integer::<Int32Type>(series, scale),
+        DataType::Int64 => laplace_impl_integer::<Int64Type>(series, scale),
+        DataType::Float32 => laplace_impl_float::<Float32Type>(series, scale),
+        DataType::Float64 => laplace_impl_float::<Float64Type>(series, scale),
+        _ => Err(PolarsError::ComputeError(
+            "Expected numeric data type.".into(),
```

Comment:
If a user had tried to use u8 or u16, and go a message "Expected numeric data type", they might be confused. Rather than making this message more complicated, add matches for u8 and u16 with a more precise error?

Reply:
Added another catch. Thx!